### ISSUE TITLE
Remove tray dependency

### ISSUE THREAD

**Issue by FFY00** [pwr-Solaar/Solaar#593]
Right now the only way to launch Solaar is by clicking in the tray icon, this is not optimal. We would want to enable users to launch Solaar directly.

**Comment by pfps** [comment]
I can probably work on this.

**Comment by FFY00** [comment]
That would be awesome! It should not be that hard I imagine, but I don't have time to do it.

**Comment by pfps** [comment]
I found what needs to be done to start with the window up (and also have the tray).  The change is simple  But that should the overall behaviour be?

One way to go would be to have a flag that selects whether to have the window open or not, something like
solaar --visible
to start with a visible window
or
solaar --tray
to start with only the tray icon.
The first is closest to current behavior, but the second seems more logical.

PS:  Have my documentation changes been pulled yet?

**Comment by pfps** [comment]
https://github.com/pwr-Solaar/Solaar/pull/611 adds --tray as a option.  It does change default behaviour.  
I haven't updated the documentation yet - I'm waiting for my other documentation changes to be pulled.

**Comment by mickleby** [comment]
@FFY00 "the only way to launch Solaar is by clicking in the tray icon" relates to the current design requirement of a single instance. The DBus design should facilitate virtually any number of solaar instances.

However, I've written many singleton apps. Something like: instance.new checks for other instance, signals instance.singleton to take focus, instance.new exits

**Comment by FFY00** [comment]
Well, it shouldn't be a problem unless the two instances are trying to communicate with the device at the same time. But this is the same problem as having Solaar and libratbag running at the same time. Actually, since the udev rules make the hidraw node accessible to everyone, any app might try to communicate with the device. The plan is to have a daemon, running as root or in its own user.

**Comment by mickleby** [comment]
My view is similar. The current design has a variety of expectations making solaar quite brittle. In this case for instance, does the GUI notice that settings change on the device? I believe solaar pulls config at launch, pushes these config, and thereafter assumes all changes occur via it. In the future DBus perhaps we create "device.listener" or "receiver.listener" that will allow notify msgs to propagate. In my example, said solaar instance subscribes to notifications of external events; perhaps re-read from device, perhaps re-read from device and push config changes, depends.

**Comment by pfps** [comment]
Based on a suggestion in #662 and some thinking about how a non-tray version of Solaar might work, I have fiddled with my initial attempt to address this problem in merged PR #611 
The basic idea is to allow Solaar to start up in three different ways:
1/ the old normal way, with a tray icon and the main window hidden
2/ the way I added in PR #611 (and made the default), with a tray icon and the main window showing
3/ with no tray icon at all and (of course) the main window showing
This is controlled by a single option --window, with three values hide, show, and only, corresponding to the three behaviours above, with the default being hide (so the default behaviour goes back to the old normal way).  The --tray option is no longer needed so I removed it.
If there is a tray icon, closing the main window just hides it.  If there is no tray icon, closing the main window quits Solaar (by using the same function that the tray menu Quit uses).

I'm looking for comments on whether this is a better solution to this issue.

I have everything above working, but I need to do a check for any cleanup that might be warranted.  If not much cleanup is needed, I'll put in a PR shortly.

**Comment by pfps** [comment]
@mickleby I'm pretty sure that the Solaar GUI only allows one instance of the program to be active at one time, although I'm not sure exactly how this is done.  The CLI doesn't have this check.

**Comment by mickleby** [comment]
> the Solaar GUI only allows one instance of the program to be active at one time, although I'm not sure exactly how this is done

Quite simple. `Solaar/lib/solaar/ui/__init__.py`
```
	APP_ID = 'io.github.pwr.solaar'
	application = Gtk.Application.new(APP_ID, 0) # global static APP_ID
```
GtkApplication wraps the singleton logic, allowing a single instance per APP_ID https://developer.gnome.org/gtk3/stable/gtk-migrating-unique-GtkApplication.html

**Comment by pfps** [comment]
PR #666 should be sufficient to close this issue.

**Comment by pfps** [comment]
Closed by PR #666
